### PR TITLE
Fix re-auth on updating profile email/password

### PR DESCRIPTION
**Pull request by transient-jonas**
Issue: [Notion Task 🦾 ](https://www.notion.so/29k/Early-Access-2794500652b34c64b0aff0dbbc53e0ab?pvs=4#4dfb97222a3a444493f5c3789faf7f34)

### Description of the Change

Changing email and/or password requires a recent login (like deleting account) 

#### Screenshots
|  |  |  |
| ------ | ------ | ------ |
|  |   |  |

### CODE REVIEW

**Review by swemail — APPROVED** [review]

**Review by gewfy — APPROVED** [review]
Really nice 👌 Just to double check, if you change your display name it doesn't require password? Only some failing unit testst :)

File: client/src/routes/modals/ProfileSettingsModal/ProfileSettingsModal.tsx [hunk 1]
```diff
@@ -142,6 +150,18 @@ const ProfileSettingsModal = () => {
               onChangeText={setEmail}
               defaultValue={email}
             />
+            {user && !user?.isAnonymous && (
+              <BottomSheetActionTextInput
+                textContentType="currentPassword"
+                secureTextEntry
+                autoCapitalize="none"
+                autoComplete="password-current"
```
Comment (gewfy) [inline client/src/routes/modals/ProfileSettingsModal/ProfileSettingsModal.tsx:158]: Oh I've done this too and we got errors in Sentry about it. If I remember correctly this is not available until RN 0.71. I think it needs to be `password` only https://reactnative.dev/docs/0.69/textinput#autocomplete-android

File: client/src/routes/modals/ProfileSettingsModal/ProfileSettingsModal.tsx [hunk 2]
```diff
@@ -142,6 +150,18 @@ const ProfileSettingsModal = () => {
               onChangeText={setEmail}
               defaultValue={email}
             />
+            {user && !user?.isAnonymous && (
+              <BottomSheetActionTextInput
+                textContentType="currentPassword"
```
Comment (gewfy) [inline client/src/routes/modals/ProfileSettingsModal/ProfileSettingsModal.tsx:155]: And this should be `password` I beleive

File: client/src/routes/modals/ProfileSettingsModal/ProfileSettingsModal.tsx (hunk 1 above)
Comment (transient-jonas) [reply client/src/routes/modals/ProfileSettingsModal/ProfileSettingsModal.tsx:158]: aha, so I don't define it for the "newPassword" field I guess?

File: client/src/routes/modals/ProfileSettingsModal/ProfileSettingsModal.tsx (hunk 2 above)
Comment (transient-jonas) [reply client/src/routes/modals/ProfileSettingsModal/ProfileSettingsModal.tsx]: updated it 👍

File: client/src/routes/modals/ProfileSettingsModal/ProfileSettingsModal.tsx (hunk 1 above)
Comment (transient-jonas) [reply client/src/routes/modals/ProfileSettingsModal/ProfileSettingsModal.tsx]: Updated 👍 
I left the new password filed without the autocomplete field

**Pull request merged**